### DISCUSSION TITLE
Please unlink the click off automatically changing the tempo if recording

### DISCUSSION THREAD

**Discussion by jimmydeer** [helgoboss/helgobox#1291] [Ideas]
This feature is really driving me nuts. I don't like hearing the metronome once I start recording (always use beats), so I turn it off after I record something to hearback without a click. hit record again (commonsense reflex from any other daw) to stop bc no count-in, the tempo resets to whatever the clip length was before I deleted it and I have to do several undo steps-workflow vibe assassin

This seems like a somewhat specialized feature (I would never use) so could it be moved somewhere else? 

I think stuff like changing the recording time should be on top menus instead of 3 clicks in and something like this could go elsewhere

Or any easy fix would be to have a "only click for preroll" option like the metronome in reaper

I'm rusty as I haven't used this for a bit and having to relearn everything reminds me it could be easier to get going

**Comment by helgoboss** [comment]
Understand. So your scenario is that you want the metronome during count-in but at zero you want it to stop because some other clip already provides the tempo reference?

**Reply by jimmydeer** [reply]
Precisely, thanks.

**Comment by helgoboss** [comment]
Okay. I think an "only click during count-in phase" solution makes sense, no matter what. Making "tempo detection recordings" not the default is probably also a good idea. After the final release, I'm planning to add a menu to the smart-record button for picking various recording modes. Maybe I could move the tempo detection option in there.

**Reply by jimmydeer** [reply]
Great thanks! 

Record button mode menu is an excellent idea-intuitive place to have options

**Comment by helgoboss** [comment]
@jimmydeer This will be in version 2.16.15.

I have thought about this quite much and implemented it a bit different than initially intended:

## Tempo detection is still the default

Doing tempo detection when the metronome is off is still the default behavior. That helps users to discover this feature, and there's a tutorial video already out there which would turn invalid if I'd change the default.

## You can turn it off

However, you can now **turn off this special behavior** in the matrix recording settings:

<img width="264" alt="image" src="https://github.com/user-attachments/assets/aae38a7b-fe76-445b-8043-69302ec1964e">

<img width="264" alt="image" src="https://github.com/user-attachments/assets/2d021ea0-aaca-44e4-87d9-9faf0929537a">

## There's guidance on how to turn it off

Moving the mouse over the red text in the title bar "If you record a clip now, Playtime will initiate a tempo detection recording!" reveals instructions how this behavior can be turned off:

![image](https://github.com/user-attachments/assets/28cd2553-a640-42f5-acbc-310a227219b4)

## The setting is instance-specific, not global

As most other non-appearance/performance-related settings, this new "Allow tempo detection recordings" setting is **not global**! @jimmydeer For you, this is inconvenient at first sight, because you never want tempo detection recordings. However, in the same release I'm going to introduce a streamlined way to use Playtime templates, see #1348.

## What about "Click only during count-in phase"?

I thought a lot about this as well and came to the conclusion that it's best to **always** click during count-in recordings, even if the metronome is not on. A count-in recording without metronome doesn't really make sense.

In addition, I have opened idea #1349, which is about introducing the possibility to click only during recording but not during playback. Not something I strictly need, but up for discussion.

**Comment by helgoboss** [comment]
Implemented in #1339.
